proxy/interceptor.go

```diff
@@ -0,0 +1,1632 @@
+				o.GetWorkflowTask(),
+			); err != nil {
+				return err
+			}
+
+		case *workflowservicepb.RespondWorkflowTaskFailedRequest:
+
+			if o == nil {
+				continue
+			}
+			if err := visitPayloads(
+				ctx,
+				options,
+				o.GetFailure(),
+			); err != nil {
+				return err
+			}
+
+		case *workflowservicepb.ScanWorkflowExecutionsResponse:
+
+			if o == nil {
+				continue
+			}
+			if err := visitPayloads(
+				ctx,
+				options,
+				o.GetExecutions(),
+			); err != nil {
+				return err
+			}
+
+		case *workflowservicepb.SignalWithStartWorkflowExecutionRequest:
+
+			if o == nil {
+				continue
+			}
+			if err := visitPayloads(
+				ctx,
+				options,
+				o.GetHeader(),
+				o.GetInput(),
+				o.GetMemo(),
+				o.GetSearchAttributes(),
+				o.GetSignalInput(),
+			); err != nil {
+				return err
+			}
+
+		case *workflowservicepb.SignalWorkflowExecutionRequest:
+
+			if o == nil {
+				continue
+			}
+			if err := visitPayloads(
+				ctx,
+				options,
+				o.GetHeader(),
+				o.GetInput(),
+			); err != nil {
+				return err
+			}
+
+		case *workflowservicepb.StartBatchOperationRequest:
+
+			if o == nil {
+				continue
+			}
+			if err := visitPayloads(
+				ctx,
+				options,
+				o.GetSignalOperation(),
+				o.GetTerminationOperation(),
+			); err != nil {
+				return err
+			}
+
+		case *workflowservicepb.StartWorkflowExecutionRequest:
+
+			if o == nil {
+				continue
+			}
+			if err := visitPayloads(
+				ctx,
+				options,
+				o.GetHeader(),
+				o.GetInput(),
+				o.GetMemo(),
+				o.GetSearchAttributes(),
+			); err != nil {
+				return err
+			}
+
+		case *workflowservicepb.TerminateWorkflowExecutionRequest:
+
+			if o == nil {
+				continue
+			}
+			if err := visitPayloads(
+				ctx,
+				options,
+				o.GetDetails(),
+			); err != nil {
+				return err
+			}
+
+		case *workflowservicepb.UpdateScheduleRequest:
+
+			if o == nil {
+				continue
+			}
+			if err := visitPayloads(
+				ctx,
+				options,
+				o.GetSchedule(),
+			); err != nil {
+				return err
+			}
+
+		case *workflowservicepb.UpdateWorkflowRequest:
+
+			if o == nil {
+				continue
+			}
+			if err := visitPayloads(
+				ctx,
+				options,
+				o.GetInput(),
+			); err != nil {
+				return err
+			}
+
+		case *workflowservicepb.UpdateWorkflowResponse:
+
+			if o == nil {
+				continue
+			}
+			if err := visitPayloads(
+				ctx,
+				options,
+				o.GetOutput(),
+			); err != nil {
+				return err
+			}
+
+		}
+	}
+
+	return nil
+}
+
+func NewPayloadVisitorInterceptor(options PayloadVisitorInterceptorOptions) (grpc.UnaryClientInterceptor, error) {
+	return func(ctx context.Context, method string, req, response interface{}, cc *grpc.ClientConn, invoker grpc.UnaryInvoker, opts ...grpc.CallOption) error {
+		err := VisitPayloads(ctx, req, *options.Outbound)
+		if err != nil {
+			return err
+		}
+
+		err = invoker(ctx, method, req, response, cc, opts...)
+		if err != nil {
+			return err
+		}
+
+		return VisitPayloads(ctx, response, *options.Inbound)
```

Comment:
Let's skip this visit if response is not proto.Message or if options.Inbound is nil